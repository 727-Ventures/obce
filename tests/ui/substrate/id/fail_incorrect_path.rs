use obce::id;

mod nested {
    #[obce::definition]
    pub trait Trait {
        #[obce(id = "named-extension-method")]
        fn extension_method(&self);
    }
}

fn main() {
    id!(nested::Trait::extension_method);
}
