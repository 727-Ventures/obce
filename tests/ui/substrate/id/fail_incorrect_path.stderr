error: id macro supports only two-segment paths (ChainExtension, ChainExtension::method)
  --> tests/ui/substrate/id/fail_incorrect_path.rs:12:9
   |
12 |     id!(nested::Trait::extension_method);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
